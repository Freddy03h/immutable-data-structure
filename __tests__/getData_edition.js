import Immutable from 'immutable'
import {
  initialState,
  createMergeRecords, createMergeCompleteListsRecords, createUpdateRecord,
  getIdsByTwoForeignId, getDataByTwoForeignId,
} from '../src/index'

import serieDeathNoteJSON from '../__fixtures__/serie_death_note.json'
import serieFullmetalAlchemistJSON from '../__fixtures__/serie_fullmetal_alchemist.json'

import {
  VolumeRecord, volumeForeignKeys,
  EditionRecord, editionForeignKeys,
  SerieRecord, serieForeignKeys,
} from '../__fixtures__/records'

const updateSerieRecord = createUpdateRecord(SerieRecord, serieForeignKeys)
const mergeVolumeRecords = createMergeRecords(VolumeRecord, volumeForeignKeys)
const mergeVolumeRecordsFromSeries = createMergeCompleteListsRecords(mergeVolumeRecords, 'volumes', 'edition_id')
const mergeEditionRecords = createMergeRecords(EditionRecord, editionForeignKeys)

const serieA = Immutable.fromJS(serieFullmetalAlchemistJSON)
const store_tmp = Immutable.Map({
  series: updateSerieRecord(initialState, serieA),
  editions: mergeEditionRecords(initialState, serieA.get('editions')),
  volumes: mergeVolumeRecordsFromSeries(initialState, serieA.get('editions')),
})

const serieB = Immutable.fromJS(serieDeathNoteJSON)
const store = Immutable.Map({
  series: updateSerieRecord(store_tmp.get('series'), serieB),
  editions: mergeEditionRecords(store_tmp.get('editions'), serieB.get('editions')),
  volumes: mergeVolumeRecordsFromSeries(store_tmp.get('volumes'), serieB.get('editions')),
})

describe('getData edition', () => {

  test('get edition ids with series_id and publisher_id', () => {
    expect(
      getIdsByTwoForeignId(store, 'editions', 'series_id', '320071be-4196-402b-98d7-d34bec8a1aab', 'publisher_id', '4c9547ff-2ef6-439a-80b8-ea705a385b76')
    ).toMatchSnapshot()
  })

  test('get edition with series_id and publisher_id', () => {
    expect(
      getDataByTwoForeignId(store, 'editions', 'series_id', '320071be-4196-402b-98d7-d34bec8a1aab', 'publisher_id', '4c9547ff-2ef6-439a-80b8-ea705a385b76')
    ).toMatchSnapshot()
  })

})
