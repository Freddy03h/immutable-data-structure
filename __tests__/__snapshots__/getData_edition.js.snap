// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData edition get edition ids with series_id and publisher_id 1`] = `
Immutable.Set [
  "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
  "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
]
`;

exports[`getData edition get edition with series_id and publisher_id 1`] = `
Immutable.Map {
  908612fd-1cc7-4780-aea4-63d3ac04c6bd: Object {
    "id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
    "publisher_id": "4c9547ff-2ef6-439a-80b8-ea705a385b76",
    "series_id": "320071be-4196-402b-98d7-d34bec8a1aab",
    "title": "Black Edition",
  },
  5482eba9-2773-40d8-82f7-1870b1b2a3e9: Object {
    "id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
    "publisher_id": "4c9547ff-2ef6-439a-80b8-ea705a385b76",
    "series_id": "320071be-4196-402b-98d7-d34bec8a1aab",
    "title": "Edition simple",
  },
}
`;
